source/image/ImageSet.cpp

```diff
@@ -153,11 +149,8 @@ namespace {
 // Check if the given path is to an image of a valid file type.
 bool ImageSet::IsImage(const string &path)
 {
-	if(path.length() < 4)
-		return false;
-
-	string ext = path.substr(path.length() - 4);
-	return (ext == ".png" || ext == ".jpg" || ext == ".PNG" || ext == ".JPG");
+	string extension = filesystem::path(path).extension().string();
```

Comment:
Using std::filesystem here but not in other places is inconsistent, use it for all cases where you get the extension or not at all.

Reply:
I can rewrite the relevant portions of ImageSet/ImageBuffer to use filesystem, but those changes might get too big. Would you be fine with that?

Reply:
> I can rewrite the relevant portions of ImageSet/ImageBuffer to use filesystem, but those changes might get too big. Would you be fine with that?

I think you should keep the current way of getting the extension for this PR, and open an issue instead for a more wholistic filesystem refactor.